lib/election.js

```diff
@@ -155,9 +154,16 @@ api.getBlockElectors = (ledgerNode, blockHeight, callback) => {
             return aPreviousElector - bPreviousElector;
           }
 
-          // 3. sort by hex hash(manifest + voter.id)
-          // TODO: cache hashes to avoid recomputation
-          return _sha256(a.id).localeCompare(_sha256(b.id));
+          // generate and cache hashes
+          // the hash of the previous block is combined with the elector id to
+          // prevent any elector from *always* being sorted to the top
+          a.hash = a.hash || _sha256(
+            results.latestBlock.eventBlock.meta.blockHash + a.id);
```

Comment:
Let's do:

```js
a.hash = a.hash || _sha256(results.latestBlock.eventBlock.meta.blockHash + _sha256(a.id));
```

Because it will ensure proper boundaries, etc.

Reply:
@dlongley Can you say more about why you think this is a good idea?

Reply:
When concatenating different values to produce a hash it's a good idea to make it clear where the boundaries are between the values, particularly if the data is user controlled, so that someone can't manipulate that data to cause confusion. It may be fairly unlikely to happen particularly with our current hash choices for blocks and the fact that the ID comes last. However, I'd rather use a best practice here out of an abundance of caution as the ID can be anything.

It's more of a problem in situations like this:

hash(myName + myFavoriteAnimal)

Where you can have:

myName = alfred
myFavoriteAnimal = fox

And:

myName = alf
myFavoriteAnimal = redfox

Reply:
@dlongley done, ready for another pass.